### PR TITLE
feat: pkg-util-github-repo enhancement

### PR DESCRIPTION
**Pull request by jxs1211**
Signed-off-by: xian-jie.shen <327411586@qq.com>

## Pre-Checklist

Note: please complete **_ALL_** items in the following checklist.

- [x] I have read through the [CONTRIBUTING.md](https://github.com/devstream-io/devstream/blob/main/CONTRIBUTING.md) documentation.
- [x] My code has the necessary comments and documentation (if needed).
- [x] I have added relevant tests

## Description
add ut for repo

## Related Issues
see https://github.com/devstream-io/devstream/issues/833 for the details

## New Behavior (screenshots if needed)
```sh
[going@dev github]$ go tool cover -func=coverage.out
github.com/devstream-io/devstream/pkg/util/github/repo.go:12:                   CreateRepo                              100.0%
github.com/devstream-io/devstream/pkg/util/github/repo.go:28:                   DeleteRepo                              100.0%
github.com/devstream-io/devstream/pkg/util/github/repo.go:51:                   GetRepoDescription                      81.8%
github.com/devstream-io/devstream/pkg/util/github/repo.go:79:                   PushLocalPathToBranch                   26.7%
github.com/devstream-io/devstream/pkg/util/github/repo.go:105:                  InitRepo                                100.0%
[going@dev github]$ 
```

### CODE REVIEW

**Comment by iyear** [comment]
@jxs1211 Please check the ci error and try to resolve the conflict. Thanks!

**Comment by jxs1211** [comment]
> @jxs1211 Please check the ci error and try to resolve the conflict. Thanks!

@iyear the conflicts have been resolved. But the ci error haven't yet, I find the output of the error:
```sh
Running Suite: GitHub Suite - /home/runner/work/devstream/devstream/pkg/util/github
===================================================================================
Random Seed: 1658982331

Will run 38 of 38 specs
••••••<nil>
••••••••••••••••••••2022-07-28 04:25:31 ℹ [INFO]  Prepare to create an organization repository: or/r
•2022-07-28 04:25:31 ℹ [INFO]  Prepare to create an organization repository: or/r
•2022-07-28 04:25:31 ℹ [INFO]  r%!(EXTRA string=mb)
Error: -28 04:25:31 !! [ERROR]  Failed to create repo: POST http://127.0.0.1:37449/api-v3/user/repos: 404  [].
•2022-07-28 04:25:31 ℹ [INFO]  r%!(EXTRA string=mb)
2022-07-28 04:25:31 ℹ [INFO]  Prepare to create an organization repository: or/r
2022-07-28 04:25:31 ℹ [INFO]  The repo r has been created.
•2022-07-28 04:25:31 ℹ [INFO]  r%!(EXTRA string=mb)
2022-07-28 04:25:31 ℹ [INFO]  Prepare to create an organization repository: or/r
2022-07-28 04:25:31 ℹ [INFO]  The repo r has been created.
•••2022-07-28 04:25:31 ℹ [INFO]  GitHub repo  was not found. Nothing to do here.
•2022-07-28 04:25:31 ✔ [SUCCESS]  GitHub repo r removed.
•••panic: test timed out after 10m0s

goroutine 170 [running]:
testing.(*M).startAlarm.func1()
	/opt/hostedtoolcache/go/1.18.4/x64/src/testing/testing.go:2029 +0x8e
created by time.goFunc
	/opt/hostedtoolcache/go/1.18.4/x64/src/time/sleep.go:176 +0x32
```
I noticed that the test time out was set to 10m which is supposed not to raise this timeout error, I've verified the test will pass successfully by setting the timeout to 10m\10s in my local environment. Only one condition can reproduce the error by setting the time out to 1s below:
```sh
[going@dev github]$ go test -run=TestPlanmanager -cover  -coverprofile=./coverage.out -timeout=1s -v
=== RUN   TestPlanmanager
Running Suite: GitHub Suite - /d/work/src/go/open_source/devstream/pkg/util/github
==================================================================================
Random Seed: 1658985340

Will run 38 of 38 specs
•••••2022-07-28 13:15:40 ℹ [INFO]  Prepare to create an organization repository: or/r
•2022-07-28 13:15:40 ℹ [INFO]  Prepare to create an organization repository: or/r
•2022-07-28 13:15:40 ℹ [INFO]  r%!(EXTRA string=mb)
2022-07-28 13:15:40 !! [ERROR]  Failed to create repo: POST http://127.0.0.1:37331/api-v3/user/repos: 404  [].
•2022-07-28 13:15:40 ℹ [INFO]  r%!(EXTRA string=mb)
2022-07-28 13:15:40 ℹ [INFO]  Prepare to create an organization repository: or/r
2022-07-28 13:15:40 ℹ [INFO]  The repo r has been created.
•2022-07-28 13:15:40 ℹ [INFO]  r%!(EXTRA string=mb)
2022-07-28 13:15:40 ℹ [INFO]  Prepare to create an organization repository: or/r
2022-07-28 13:15:40 ℹ [INFO]  The repo r has been created.
•••2022-07-28 13:15:40 ℹ [INFO]  GitHub repo  was not found. Nothing to do here.
•2022-07-28 13:15:40 ✔ [SUCCESS]  GitHub repo r removed.
••••••••••••••••••••••<nil>
•coverage: 55.9% of statements
panic: test timed out after 1s

goroutine 173 [running]:
testing.(*M).startAlarm.func1()
        /home/going/go/go1.18.3/src/testing/testing.go:2029 +0x8e
```


so I'm a little bit confused why this error could happened, or there is something I missed?

File: pkg/util/github/repo_test.go [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+)
+
+var _ = Describe("Repo", func() {
+	var owner, repo, org = "o", "r", "or"
+	var rep *go_github.Repository
+	defaultBranch := "db"
+	mainBranch := "mb"
+	filePath := ".placeholder"
+
+	Context(("do AddRepoSecret"), func() {
```
Comment (aFlyBird0) [inline pkg/util/github/repo_test.go:23]: redundant `()`

File: pkg/util/github/repo_test.go [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+		It("do GetRepoPublicKey with wrong url", func() {
```
Comment (aFlyBird0) [inline pkg/util/github/repo_test.go:24]: do -> does

Comment (jxs1211) [reply pkg/util/github/repo_test.go:24]: updated

File: pkg/util/github/pullrequest_test.go [hunk 3]
```diff
@@ -83,7 +83,7 @@ var _ = Describe("Pullrequest", func() {
 		})
 	})
 
-	Context(("do MergePullRequest"), func() {
+	Context("do MergePullRequest", func() {
 
 		BeforeEach(func() {
 			mux.HandleFunc("/repos/or/r/pulls/1/merge", func(w http.ResponseWriter, r *http.Request) {
```
[the remark this answers is not in the document]
Comment (jxs1211) [reply pkg/util/github/pullrequest_test.go:89]: good to know that

[the remark this answers is not in the document]
Comment (jxs1211) [reply pkg/util/github/pullrequest_test.go:89]: @steinliber take your advice and refactor the code

**Comment by jxs1211** [comment]
> @jxs1211 Please check the ci error and try to resolve the conflict. Thanks!

@iyear I've rewritten the code and fixed the error.

**Review by iyear — APPROVED** [review]
Awesome!

**Pull request merged**